### PR TITLE
Handle MDN links correctly

### PR DESCRIPTION
**Pull request by kumar303**
Fixes https://github.com/mozilla/addons-code-manager/issues/383

Example:

Screenshot 2019-03-20 14 50 44

### CODE REVIEW

File: src/components/LinterMessage/index.tsx [hunk 1]
```diff
@@ -32,20 +33,24 @@ const renderDescription = (description: LinterMessage['description']) => {
...
-            <Alert.Link key={`link:${part}`} href={part}>
-              {part}
-            </Alert.Link>,
-          );
-        } else {
-          allParts.push(decodeHtmlEntities(part));
-        }
+      line
+        // Replace anchor tags with just their URLs.
+        .replace(linkPattern, '$1')
```
Comment (willdurand) [inline src/components/LinterMessage/index.tsx]: Why not using DOMPurify (already used in this project) to remove all HTML tags (given that we don't intend to render HTML)? The regexp seems fragile if, for instance, new attributes are added (e.g., `target` and `rel`).

Comment (kumar303) [reply src/components/LinterMessage/index.tsx]: It seems unlikely to me that this html would change but I could try to see if dompurify is easy enough to work with. It probably won't give me the `href` but maybe that's OK.

Comment (willdurand) [reply src/components/LinterMessage/index.tsx]: oh wait, right. it won't work well if the `href` value is different than the link content.

File: src/components/LinterMessage/index.tsx [hunk 2]
```diff
@@ -24,6 +24,7 @@ export const decodeHtmlEntities = (text: string) => {
 };
 
 const renderDescription = (description: LinterMessage['description']) => {
+  const linkPattern = /<a href="([^"]+)">[^<]+<\/a>/g;
```
Comment (willdurand) [inline src/components/LinterMessage/index.tsx]: should we make it case insensitive too?

Comment (willdurand) [reply src/components/LinterMessage/index.tsx]: maybe you could try to match a link with attributes, just in case. WDYT?

**Comment by kumar303** [comment]
Good call on dompurify. I think that's a safe route to go. The anchor text will always be a URL (as far as I know) so it should work fine. If a reviewer notices a problem, they will tell us.

File: src/components/LinterMessage/index.tsx [hunk 3]
```diff
@@ -32,20 +33,24 @@ const renderDescription = (description: LinterMessage['description']) => {
...
-            <Alert.Link key={`link:${part}`} href={part}>
-              {part}
-            </Alert.Link>,
-          );
-        } else {
-          allParts.push(decodeHtmlEntities(part));
-        }
+      // Replace anchor tags with just their text, which are the URLs.
+      purify
+        .sanitize(line, { ALLOWED_TAGS: [] })
```
Comment (willdurand) [inline src/components/LinterMessage/index.tsx]: 🏆

**Pull request merged**